src/xdp/trn_trace_prog_kern.c

```diff
@@ -0,0 +1,77 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+/**
+ * @file trn_trace_prog_kern.c
+ * @author Sherif Abdelwahab (@zasherif)
+ *         Phu Tran          (@phudtran)
+ *
+ * @brief Implements the XDP monitoring program (metrics collector)
+ *
+ * @copyright Copyright (c) 2020 The Authors.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 2 of the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ */
+
+#include <linux/bpf.h>
+#include <bpf/bpf_helpers.h>
+
+#include "bpf_legacy.h"
+
+#include "trn_datamodel.h"
+#include "trn_transit_xdp_maps.h"
+#include "trn_kern.h"
+
+int _version SEC("version") = 1;
+
+SEC("trace_metrics_per_packet")
+int trace_metrics_per_packet(int *act, struct transit_packet *pkt)
```

Comment:
Do we need a section here for ebpf-to-ebpf call? Would you please change that to:
```
static __inline int trace_metrics_per_packet(int *act, struct transit_packet *pkt) {
```

Reply:
agreed

Reply:
Fixed.
And I also change the trn_trace_prog_kern.c to trn_trace_prog_kern.h and include it in trn_transit_xdp.c